### PR TITLE
Added deprecated warning to message type danger

### LINKED ISSUE #1085: [Enhancement] Remove danger (red) state of toast
**Is your enhancement request related to a problem? Please describe.**
We've decided to clean up our toast definition and flavors: Toasts are in their nature never "critical" and should thus not be red - Red should be used sparingly, only for destructive actions, and thus only in **Alerts** because they are by their nature more "critical"

**Additional context**
All red toasts should be yellow instead and green should stay green

### PR DESCRIPTION
**Pull request by bdytos**
This PR closes #1085  (reference issue number here)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Enhancement (to existing content)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Which issue documents the current behaviour?

Issue Number: #[replace with issue number]

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: libs/designsystem/src/lib/components/toast/services/toast.helper.ts [hunk 1]
```diff
@@ -30,9 +29,18 @@ export class ToastHelper {
...
     let cssClass = ToastHelper.CSS_CLASS;
-    if (messageType) {
-      cssClass += ' ' + messageType;
+    const type: MessageType =
+      messageType !== 'danger' ? messageType : this.warnAboutDepricatedMessageType();
+    if (type) {
+      cssClass += ' ' + type;
     }
     return cssClass;
   }
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/toast/services/toast.helper.ts:38]: I think we could simplify this a bit, and make it easier to just remove the extra check when we decide the deprecation period is over, like this:

```suggestion
    if (messageType === 'danger') {
      return cssClass += ' ' + 'warning';
    }
    return cssClass += ' ' + messageType;
  }
```

Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/toast/services/toast.helper.ts:38]: Then you just have to return the `cssClass += warning` in the method, or move it to the check.

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]
Looks good, added a couple of comments, mostly just an idea of how we could make it as easy as possible to deprecate down the line. 

Also, dont forget to describe this change in the cookbook [/toast-showcase/toast-showcase.component.html](https://github.com/kirbydesign/designsystem/blob/master/apps/cookbook/src/app/showcase/toast-showcase/toast-showcase.component.html)

File: libs/designsystem/src/lib/components/toast/services/toast.helper.ts (hunk 1 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/toast/services/toast.helper.ts:38]: E.g. like this:
```suggestion
    private getCssClass(messageType: MessageType): string {
    let cssClass = ToastHelper.CSS_CLASS;
    if (messageType === 'danger') {
      console.warn(
        `[DEPRECATED] 'danger' message type is deprecated. Use Alerts for critial events. Message type will be set as 'warning'`
      );
      return cssClass += ' ' + 'warning';
    }
    return cssClass += ' ' + messageType;
  }
```

File: libs/designsystem/src/lib/components/toast/config/toast-config.ts [hunk 2]
```diff
@@ -5,4 +5,7 @@ export interface ToastConfig {
   animated?: boolean;
 }
 
+/**
+ * 'danger' is deprecated. Use alert for critical events
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/toast/config/toast-config.ts:9]: Add @deprecated to beginning of message

File: apps/cookbook/src/app/examples/toast-example/toast-example.component.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export class ToastExampleComponent {
   showToast() {
     const config: ToastConfig = {
       message: 'Your toast message',
-      messageType: 'success',
+      messageType: 'danger',
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/examples/toast-example/toast-example.component.ts:15]: Please revert

File: libs/designsystem/src/lib/components/toast/config/toast-config.ts [hunk 4]
```diff
@@ -5,4 +5,7 @@ export interface ToastConfig {
   animated?: boolean;
 }
 
+/**
+ * @deprecated 'danger' is deprecated. Use alert for critical events
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/toast/config/toast-config.ts:9]: We decided to revert this, right?

File: apps/cookbook/src/app/examples/toast-example/toast-example.component.ts (hunk 3 above)
Comment (jakobe) [inline apps/cookbook/src/app/examples/toast-example/toast-example.component.ts:15]: Please revert

File: libs/designsystem/src/lib/components/toast/config/toast-config.ts [hunk 5]
```diff
@@ -5,4 +5,7 @@ export interface ToastConfig {
...
+ */
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/toast/config/toast-config.ts:10]: You shouldn't deprecate entire `MessageType` type - only the 'warning' option is deprecated

File: libs/designsystem/src/lib/components/toast/services/toast.helper.ts [hunk 6]
```diff
@@ -30,9 +29,18 @@ export class ToastHelper {
...
+        `[DEPRECATED] 'danger' message type is deprecated. Use Alerts for critial warnings. Message type will be set as 'warning'`
+      );
+
+      return (cssClass += ' ' + 'warning');
     }
-    return cssClass;
+    return (cssClass += ' ' + messageType);
+  }
+
+  private warnAboutDepricatedMessageType(): MessageType {
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/toast/services/toast.helper.ts:43]: Please remove

File: libs/designsystem/testing-base/src/lib/components/mock.range.component.ts [hunk 7]
```diff
@@ -1,34 +1,35 @@
...
-  @Output() change = new EventEmitter<number>();
-
-  setDisabledState() {}
-  writeValue() {}
-  registerOnTouched() {}
-  registerOnChange() {}
-}
-
-// #endregion
+import { Component, EventEmitter, forwardRef, Input, Output } from '@angular/core';
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/testing-base/src/lib/components/mock.range.component.ts:1]: Please revert unrelated changes

File: libs/designsystem/src/lib/components/toast/services/toast.helper.spec.ts [hunk 8]
```diff
@@ -79,10 +79,24 @@ describe('ToastHelper', () => {
           await TestHelper.whenReady(ionToast);
           const toastWrapper = ionToast.shadowRoot.querySelector('.toast-wrapper');
           expect(toastWrapper).toHaveComputedStyle({
-            'background-color': getColor(notificationColor),
+            'background-color':
+              messageType !== 'danger' ? getColor(notificationColor) : getColor('warning'),
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/toast/services/toast.helper.spec.ts:83]: Consider readability:
```suggestion
            const expectedColor = messageType === 'danger' ? getColor('warning'): getColor(notificationColor);
            'background-color': expectedColor,
```

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]
Almost there :) Please revert the unrelated changes.

File: libs/designsystem/src/lib/components/toast/services/toast.helper.spec.ts [hunk 9]
```diff
@@ -79,10 +79,24 @@ describe('ToastHelper', () => {
...
           });
         });
       });
+
+      it('Danger type should display warning in console', async () => {
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/toast/services/toast.helper.spec.ts:88]: Consider wording (tests reads like: "it should xxx....")
```suggestion
      it(`should display warning in console when using 'danger' MessageType`, async () => {
```

File: libs/designsystem/src/lib/components/toast/services/toast.helper.ts [hunk 10]
```diff
@@ -30,9 +29,18 @@ export class ToastHelper {
...
+    return 'warning';
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/toast/services/toast.helper.ts:44]: What's this? 🤔

File: libs/designsystem/src/lib/components/toast/services/toast.helper.ts [hunk 11]
```diff
@@ -30,9 +29,18 @@ export class ToastHelper {
...
+
+    if (messageType === 'danger') {
+      console.warn(
+        `[DEPRECATED] 'danger' message type is deprecated. Use Alerts for critial warnings. Message type will be set as 'warning'`
+      );
+
+      return (cssClass += ' ' + 'warning');
     }
-    return cssClass;
+    return (cssClass += ' ' + messageType);
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/toast/services/toast.helper.ts:40]: The null check for messageType is gone - why? 🤔

**Review by jakobe — CHANGES_REQUESTED** [review]

File: libs/designsystem/src/lib/components/toast/services/toast.helper.ts (hunk 11 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/toast/services/toast.helper.ts:40]: This is partly my fault, I suggested this, and dont understand why we need the null check here, when the parameter is typed?

Comment (bdytos) [reply libs/designsystem/src/lib/components/toast/services/toast.helper.ts:40]: Added the null check again

**Review by RasmusKjeldgaard — DISMISSED** [review]

File: libs/designsystem/src/lib/components/toast/services/toast.helper.spec.ts [hunk 12]
```diff
@@ -78,11 +78,26 @@ describe('ToastHelper', () => {
           const ionToast = window.document.getElementsByTagName('ion-toast')[0];
           await TestHelper.whenReady(ionToast);
           const toastWrapper = ionToast.shadowRoot.querySelector('.toast-wrapper');
+          const expectedColor =
+            messageType !== 'danger' ? getColor(notificationColor) : getColor('warning');
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/toast/services/toast.helper.spec.ts:82]: Consider readability:
```suggestion
          const expectedColor =
            messageType === 'danger' ? getColor('warning') : getColor(notificationColor);
```

File: libs/designsystem/src/lib/components/toast/config/toast-config.ts (hunk 2 above)
Comment (jakobe) [inline libs/designsystem/src/lib/components/toast/config/toast-config.ts:9]:
```suggestion
 * 'danger' is deprecated. Use Kirby Alert for critical events
```

File: libs/designsystem/src/lib/components/toast/services/toast.helper.ts [hunk 13]
```diff
@@ -30,9 +29,19 @@ export class ToastHelper {
 
   private getCssClass(messageType: MessageType): string {
     let cssClass = ToastHelper.CSS_CLASS;
+
+    if (messageType === 'danger') {
+      console.warn(
+        `[DEPRECATED] 'danger' message type is deprecated. Use Alerts for critial warnings. Message type will be set as 'warning'`
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/toast/services/toast.helper.ts:35]: Typo:
```suggestion
        `[DEPRECATED] 'danger' message type is deprecated. Use Alerts for critical warnings. Message type will be set as 'warning'`
```

Comment (jakobe) [inline libs/designsystem/src/lib/components/toast/services/toast.helper.ts:35]: Consider wording:
```suggestion
        `[DEPRECATED] 'danger' message type is deprecated. Use Alerts for critial warnings. Toast will be shown as 'warning'`
```

File: libs/designsystem/src/lib/components/toast/services/toast.helper.ts [hunk 14]
```diff
@@ -30,9 +29,19 @@ export class ToastHelper {
...
+      );
+
+      return (cssClass += ' ' + 'warning');
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/toast/services/toast.helper.ts:38]: Consider the complexity of assigning + returning here while only assigning in other case and finally returning.
Maybe just switch the `messageType` to  'warning' and keep existing null-check + return statement...

**Review by jakobe — CHANGES_REQUESTED** [review]

**Review by jakobe — APPROVED** [review]
